### ISSUE TITLE
App crashes trying to highlight a feature near a navigation endpoint

### ISSUE THREAD

**Issue by kidhanis** [organicmaps/organicmaps#9063]
⚠ Have you searched for similar, already existing issues?
Yes

**Describe the issue**
I selected a point on the map to be part of a route. Then, when I select a nearby point of interest through search, the app immediately closes instead of showing the POI.

Alternatively, tapping on a road label, which sometimes results in the blue circle showing up on another part of the road that's near the selected point for navigation, can also trigger the crash.

**Steps to reproduce**
For the crash with a POI:
1. Download map for United States Virgin Islands. I mainly chose this map because it's small (3MB) and has road labels. The crash could happen on any map.
2. Back in the world map, search for "STT". The two relevant results are the Airport POI named Cyril E King Airport and the Attraction POI named Cyril E King Airport (STT).
3. Select the Airport POI. App zooms in on airport.
4. Select "Route to". The flag is placed on the airport. At this point, the Attraction POI might be visible on the right side of the screen.
5. Zoom out three times using the zoom button on the screen. The Attraction POI should disappear at this level.
6. Tap on the search button on the left side of the screen and search for "STT" again, this time selecting the Attraction POI. App crashes.

**Expected behaviour**
App should zoom in to the selected POI and display a blue circle over it.

**Screenshots**
Snippet of the screen after step 4 showing the relevant data: the airport selected as destination, the zoom out button on the right side of the screen, the Attraction POI partially visible between the zoom buttons, and the search button on the left.
![Crash-navigation-POI](https://github.com/user-attachments/assets/637262e0-c5b2-482c-b624-ddb49da59a12)

**System information:**
 - Operating system and its version: Android 11 and 13
 - Organic Maps version: 2024.08.16-5-FDroid
 - Device Model: Samsung Galaxy S10e and Galaxy A14 5G

**Additional context**
Steps for reproducing crash using road labels:
1. Use map for United States Virgin Islands and go to same airport.
2. Zoom out until you see road labels "VI 303" and "303" north of the airport.
3. Tap on label "VI 303". The blue dot shows up not over the label but south of it, closer to the airport.
4.  Select "Route to". Flag is placed on this point, away from the label.
5. Go back to the same label "VI 303" and tap on it. App crashes.

**Comment by biodranik** [comment]
I confirm the crash with road shields. It is caused by the change in the bookmark selection code on the tap. FillBookmarkInfo is wrongly called for a route-type UserMark returned by FindBookMarkInPosition call.

@strump the code logic should be revised there. At minimum:
- FindBookMarkInPosition should be renamed to FindUserMarkInPosition
- The code starting with ` switch (mark->GetMarkType())` above should be reused for this second selection

Or, alternatively, we may revert back https://github.com/organicmaps/organicmaps/pull/8687 and try to apply some alternative approach or postpone the fix.
